ReactiveObjC/RACBlockTrampoline.m

```diff
@@ -28,7 +28,7 @@ - (instancetype)initWithBlock:(id)block {
 + (id)invokeBlock:(id)block withArguments:(RACTuple *)arguments {
 	NSCParameterAssert(block != NULL);
 
-	RACBlockTrampoline *trampoline = [[self alloc] initWithBlock:block];
+	RACBlockTrampoline *trampoline = [(RACBlockTrampoline *)[self alloc] initWithBlock:block];
```

Comment:
Why was this needed?

Reply:
While debugging the [`pod lib lint` failures on CI](https://travis-ci.org/ReactiveCocoa/ReactiveObjC/jobs/472432768) I noticed warnings like this:

```
- WARN  | [iOS] xcodebuild:  /Users/travis/build/ReactiveCocoa/ReactiveObjC/ReactiveObjC/RACBlockTrampoline.m:31:49: warning: 'initWithBlock:' is only available on iOS 10.0 or newer [-Wunguarded-availability]
```

This is a false positive, because it’s confusing our `-[RACBlockTrampoline initWithBlock:]` selector for something like `-[NSThread initWithBlock:]`, which has only been available since iOS 10. Adding the explicit cast shouldn’t be necessary (this seems like a compiler/analyser bug), but it does remove the warning.